src/main/java/lazyprogrammer/jwtdemo/services/UserService.java

```diff
@@ -21,6 +21,7 @@ public ResponseEntity<UserDto> saveUser(UserDto dto) {
         User entity = new User();
         dto.setPassword(passwordEncoder.encode(dto.getPassword()));
         BeanUtils.copyProperties(dto, entity);
+
```

Comment:
We need to fetch the List of Role objects based on Role names provided in registration request. Then add the elements of that list as a Set in User Entity object before persisting.